pkg/log/request.go

```diff
@@ -24,13 +24,13 @@ import (
 )
 
 // CorrelationIDHeaders are the headers whose values will be taken as a correlation id for incoming requests
-var CorrelationIDHeaders = []string{"X-Correlation-ID", "X-CorrelationID", "X-ForRequest-ID"}
+var CorrelationIDHeaders = []string{"X-Correlation-ID", "X-CorrelationID", "X-ForRequest-ID", "X-Request-ID"}
 
 // CorrelationIDForRequest returns checks the http headers for any of the supported correlation id headers.
 // The first that matches is taken as the correlation id. If none exists a new one is generated.
 func CorrelationIDForRequest(request *http.Request) string {
 	for key, val := range request.Header {
-		if slice.StringsAnyEquals(CorrelationIDHeaders, key) {
+		if slice.StringsAnyEqualsIgnoreCase(CorrelationIDHeaders, key) {
```

Comment:
does it make sense to add a simple test(s) to validate the header case doesn't matter?